### PR TITLE
Combine some Docker instructions for making the image smaller.

### PR DESCRIPTION
**Pull request by tmatsuo**
Also minor version updates.

The image size was ~830MB, but with this change, now it's ~650MB.

### CODE REVIEW

**Comment by cedricziel** [comment]
I'm generally fine with this. What's the virtual size of the base image?

File: php-nginx/build-scripts/build_php56.sh [hunk 1]
```diff
@@ -47,7 +47,7 @@ rm mailparse.tar.gz
 
 mkdir -p ${PHP_SRC}/ext/apcu
 # The newest 5.1.2 doesn't build with PHP 5.6.
-curl -SL "https://pecl.php.net/get/apcu-4.0.10.tgz" -o apcu.tar.gz
+curl -SL "https://pecl.php.net/get/apcu-4.0.11.tgz" -o apcu.tar.gz
```
Comment (JustinBeckwith) [inline php-nginx/build-scripts/build_php56.sh]: total nit - but shouldn't we be doing version updates in a different PR?

Comment (tmatsuo) [inline php-nginx/build-scripts/build_php56.sh]: I separated the changes

**Comment by tmatsuo** [comment]
ugh, github doesn't pick up my changes, I will delete it and open a new one

**Comment by tmatsuo** [comment]
Sorry guys the new one is here:
https://github.com/GoogleCloudPlatform/php-docker/pull/94

**Comment by tmatsuo** [comment]
@cedricziel Do you mean "gcr.io/google_appengine/base"? Then it's 246.1 MB

**Comment by cedricziel** [comment]
@tmatsuo Thank you for getting back to me :) 

Yes - that's a huge difference; did you spend some thoughts on the initial idea of maybe being able to split out the static compiled PHP Versions? I could imagine being able to push those as artifacts to GCS. On build, the user could decide which version of the interpreter and which version and / or flavor of http-daemon (nginx|apache|hhvm) should be pulled in.

That could make for a lot smaller images and therefore faster deployments. Not to mention the flexibility.

**Comment by raggi** [comment]
There have been proposals internally to start a new convention of providing versioned image names, for example:
`gcr.io/google_appengine/php-nginx-5.6`
We haven't gotten to do this yet, or even to ratify such a proposal, but it's been discussed quite a few times. Personally I think we should do this.

Regarding pushing artifacts to GCS and adding them on build. I'm not sure this would have exactly the effect you desire, in that, this would result in your builds generating much larger personal artifacts. You would be generating a user owned layer in your image that contains the result of the installation of that artifact.

Now, again, I have proposed and will keep pushing for us actually building the various non-standard packages as .deb files, and separating the builds of those out from the runtime Dockerfiles. This should improve many things, not least of which would be our end to end test performance for the runtime docker images and more repeatable builds at specific layers. Personally I don't think it would be ideal to have users have to install those debs themselves (unless they're using runtime:custom). Instead I would prefer to handle major versions and different options (such as nginx for php) in terms of the docker image name. Taking this approach minimizes the time taken to fetch the base runtime image during a deploy, and reduces the size of the user image layers when they're pushed after a build.

**Comment by raggi** [comment]
@JustinBeckwith Take a look at the above comment, this plays again into many of our discussions :)

**Comment by cedricziel** [comment]
Sure that's another way to put it and having different names for different versions isn't a bad idea either. I just wanted to make sure it's still on the page.
